### PR TITLE
Add binary_cross_entropy in functional.frontends.torch

### PR DESCRIPTION
**Pull request by whitepurple**
[updated] pr for https://github.com/unifyai/ivy/issues/3037
- [x] binary_cross_entropy

- add ivy/functional/frontends/torch/nn/loss_functions.py
  - implement 'binary_cross_entropy'
  
- add ivy_tests/test_ivy/test_frontends/test_torch/test_loss_functions.py
  - implement 'test_binary_cross_entropy'

### CODE REVIEW

**Comment by whitepurple** [comment]
Hello @jiahanxie353 ,
This pull request is ready for review, including formatting.
The code was written with reference to torch.nn.functional, 
and the warning that originally occurred in torch was omitted.

**Comment by whitepurple** [comment]
#3028

**Comment by whitepurple** [comment]
Hello @jiahanxie353, @1Doomdie1, @sreegandh804.
Please review or give feedback on this pr so that I can work on the next one. Thanks.

File: ivy/functional/frontends/torch/loss_functions.py [hunk 1]
```diff
@@ -16,3 +53,29 @@ def cross_entropy(
...
+    reduction = _get_reduction(reduction, size_average, reduce)
+    result = ivy.binary_cross_entropy(target, input, epsilon=0.0)
+    
+    if weight is not None:
+        result = ivy.multiply(weight, result)
+    result = reduction(result)
+    return result
+
+
+binary_cross_entropy.unsupported_dtypes = (
```
Comment (jiahanxie353) [inline ivy/functional/frontends/torch/loss_functions.py:75]: are you sure it does not support so many dtypes? did you try to delete one of them and test it?

File: ivy/functional/frontends/torch/loss_functions.py [hunk 2]
```diff
@@ -2,6 +2,43 @@
 import ivy
 
 
+def _get_reduction_func(reduction):    
+    if reduction == 'none':
+        ret = lambda x : x
+    elif reduction == 'mean':
+        ret = ivy.mean
+    elif reduction == 'elementwise_mean':
```
Comment (jiahanxie353) [inline ivy/functional/frontends/torch/loss_functions.py:10]: just curious, could you explain why did you add `elementwise_mean`? as far as i'm concerned, torch only has 3 `reduction`s

**Review by jiahanxie353 — COMMENTED** [review]
Sorry for the delay, left comments below. Looks good to me in general!

File: ivy/functional/frontends/torch/loss_functions.py (hunk 1 above)
Comment (whitepurple) [reply ivy/functional/frontends/torch/loss_functions.py:75]: Pytorch's bce does not support integer datatype targets, but ivy's bce does. 
I misunderstood that and wrote `unsupported_dtypes` incorrectly. 
I will modify and retest `unsupported_dtypes` under the assumption that the code of the torch that works normally is changed to ivy. 
The integer datatype will be deleted.

File: ivy/functional/frontends/torch/loss_functions.py (hunk 2 above)
Comment (whitepurple) [reply ivy/functional/frontends/torch/loss_functions.py:10]: `elementwise_mean` is a deprecated option for many changes in torch. 
It seems to be an option that remained as I wrote it referring to the `get_enum` of `torch.nn._reduction.py` among the torch's official code. 
It's an old deprecated setting, so it's okay to delete it. 
I will delete it to reflect the review.
The `elementwise_mean` option will be removed.

**Comment by whitepurple** [comment]
I have updated all the changes following the reviews.
The test code was verified by executing max_examples=1000 in `conftest.py`.

**Comment by jiahanxie353** [comment]
> I have updated all the changes following the reviews. The test code was verified by executing max_examples=1000 in `conftest.py`.

lgtm, I will merge it. Thanks a lot for contributing!
